From: Simon McVittie <smcv@collabora.com>
Date: Sun, 6 Sep 2020 17:45:04 +0100
Subject: xdp: Send messages to stderr, not stdout

This is done for two reasons:

1. Logging on stdout can interfere with machine-readable protocols.

2. stdout is normally fully buffered, whereas stderr is normally
   line-buffered. It's a lot easier to debug the service when messages
   don't lag behind.

Signed-off-by: Simon McVittie <smcv@collabora.com>
Forwarded: https://github.com/flatpak/xdg-desktop-portal/pull/521
---
 src/xdg-desktop-portal.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/xdg-desktop-portal.c b/src/xdg-desktop-portal.c
index f049d19..88a6c5e 100644
--- a/src/xdg-desktop-portal.c
+++ b/src/xdg-desktop-portal.c
@@ -79,9 +79,9 @@ message_handler (const gchar *log_domain,
 {
   /* Make this look like normal console output */
   if (log_level & G_LOG_LEVEL_DEBUG)
-    printf ("XDP: %s\n", message);
+    fprintf (stderr, "XDP: %s\n", message);
   else
-    printf ("%s: %s\n", g_get_prgname (), message);
+    fprintf (stderr, "%s: %s\n", g_get_prgname (), message);
 }
 
 static void
